source/base/qprojector.cc

```diff
@@ -688,8 +690,8 @@ QProjector<3>::project_to_all_faces(const ReferenceCell &     reference_cell,
             static_cast<const ScalarPolynomialsBase<2> &>(poly_quad);
 
         // ... and over all possible orientations
-        for (unsigned int orientation = 0;
-             orientation < (n_shape_functions * 2);
+        for (unsigned char orientation = 0;
+             orientation < reference_cell.n_face_orientations(face_no);
```

Comment:
You compare an unsigned char with an unsigned int now, but that is probably not problematic.
https://www.dealii.org/developer/doxygen/deal.II/classReferenceCell.html#a9eac7bdc3805d1a09db0a557f156c551

Reply:
Yes, that is ok.